### PR TITLE
vtk-m: Add sycl option to vtk-m package

### PR DESCRIPTION
**Pull request by kmorel**

### CODE REVIEW

File: var/spack/repos/builtin/packages/vtk-m/package.py [hunk 1]
```diff
@@ -162,6 +170,14 @@ class VtkM(CMakePackage, CudaPackage, ROCmPackage):
     # https://gitlab.kitware.com/vtk/vtk-m/-/merge_requests/3259
     patch("mr3259-thrust-is_arithmetic-fix.patch", when="@2.0.0:2.2.0 +cuda ^cuda@12.6:")
 
+    # VTK-m PR#3271
+    # https://gitlab.kitware.com/vtk/vtk-m/-/merge_requests/3271
+    patch("mr3271-contourtree-print-error.patch", when="@2.0:2.2")
+
+    # VTK-m PR#3272
+    # https://gitlab.kitware.com/vtk/vtk-m/-/merge_requests/3272
+    patch("mr3272-bad-mir-table-method.patch")
```
Comment (vicentebolea) [inline var/spack/repos/builtin/packages/vtk-m/package.py]:
```suggestion
    patch("mr3272-bad-mir-table-method.patch", when="@2:")
```

**Review by vicentebolea — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/vtk-m/package.py (hunk 1 above)
Comment (kmorel) [reply var/spack/repos/builtin/packages/vtk-m/package.py]: Good catch. I pushed a fix.

**Review by vicentebolea — APPROVED** [review]
Looks good! Thanks

**Comment by kmorel** [comment]
Thanks, @vicentebolea.

It looks like I still missing an approving review. Do you know how I can get that? This is my first actual spack submission.

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]
Edit: @kmorel Thank you very much for the explanation, and I'm sorry that I tried incorrect specs!

**Comment by kmorel** [comment]
@bernhardkaindl Can you explain to a newbie where these specs are defined? Neither is a valid spec for the vtk-m package (nor were either valid before these changes). There is no way these passed any former version of the vtk-m package, and I have not asked for a spec of this nature anywhere.

**Review by bernhardkaindl — APPROVED** [review]
As the maintainer @vicentebolea approved, I'm ready to approve too.

**Comment by kmorel** [comment]
Thanks for the help, @bernhardkaindl.

One hopefully finally question: What is left to merge in these changes. It looks like the merge is now waiting on some checks. Some are skilled and one says it is disabled for draft PRs. But, this PR is not marked as draft and I don't think I have permissions to launch any CI myself.

**Comment by bernhardkaindl** [comment]
@spackbot run pipeline

**Comment by kmorel** [comment]
Awesome! Thanks.

**Pull request merged**